Src/BootCamp.Chapter/BinaryConverter.cs

```diff
@@ -1,19 +1,82 @@
 ﻿using System;
-using System.Collections.Generic;
-using System.Text;
 
 namespace BootCamp.Chapter
 {
     public static class BinaryConverter
     {
         public static long ToInteger(string binary)
         {
-            return 0;
+            if (String.IsNullOrEmpty(binary))
+            {
+                return 0L;
+            }
+            var isValid = IsBin(binary);
+            if (!isValid)
+            {
+                throw new InvalidBinaryNumberException(binary);
+            }
+            var number = int.Parse(binary);
+            int convertedNumber = 0;
+            int base1 = 1;
+
+            while (number > 0)
+            {
+                int remainder = number % 10;
+                number /= 10;
+                convertedNumber += remainder * base1;
+                base1 *= 2;
+            }
+            return convertedNumber;
+        }
+
+        public static bool IsBin(string s)
+        {
+            foreach (var c in s)
+                if (c != '0' && c != '1')
+                    return false;
+            return true;
         }
 
         public static string ToBinary(long number)
         {
-            return "";
+            try
+            {
+                var realNumber = number;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove the unused local variable 'realNumber'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4977576)

Reply:
solved